### PR TITLE
[DOCFIX] Update tuning docs

### PR DESCRIPTION
**Pull request by yuzhu**
### What changes are proposed in this pull request?

Doc update for tuning 

### Why are the changes needed?

Help users to maximize Alluxio benefit

### Does this PR introduce any user facing changes?
Doc fix

### CODE REVIEW

**Comment by yuzhu** [comment]
@jiacheliu3 @sg-c PTAL, I tried to update some sections of the tuning guide.

File: docs/en/operation/Performance-Tuning.md [hunk 1]
```diff
@@ -214,6 +214,19 @@ Increasing the number of threads can decrease the staleness of the UFS path cach
...
+
+If the content on the UFS are modified without going through Alluxio, Alluxio needs to sync its metadata with the UFS to reflect those changes in Alluxio namespace. 
+The cost of metadata sync scales linearly with the number of files in the directory that is being synced. 
+If metadata sync operation happens frequently on large directories, more threads may be allocated to speed up this process. 
+Two configurations are relevant here. 
+
+'alluxio.master.metadata.sync.concurrency.level' controls the concurrency that is used in a single sync operation.
+Adjust this to 1x to 2x core count on the master node to speed up the speed of metadata sync.
+
+'alluxio.master.metadata.sync.executor.pool.size' controls the number of threads performing sync operations.
```
Comment (jiacheliu3) [inline docs/en/operation/Performance-Tuning.md:227]: Do you wanna mention the prefetch thread pool?

File: docs/en/operation/Performance-Tuning.md [hunk 2]
```diff
@@ -214,6 +214,19 @@ Increasing the number of threads can decrease the staleness of the UFS path cach
...
 If this is set to 0, the cache is disabled and the `ONCE` setting will behave like the `ALWAYS` setting.
 
+### Metadata Sync
+
+If the content on the UFS are modified without going through Alluxio, Alluxio needs to sync its metadata with the UFS to reflect those changes in Alluxio namespace. 
+The cost of metadata sync scales linearly with the number of files in the directory that is being synced. 
+If metadata sync operation happens frequently on large directories, more threads may be allocated to speed up this process. 
+Two configurations are relevant here. 
+
+'alluxio.master.metadata.sync.concurrency.level' controls the concurrency that is used in a single sync operation.
```
Comment (jiacheliu3) [inline docs/en/operation/Performance-Tuning.md:224]: nit: will backtick be better?

File: docs/en/operation/Scalability-Tuning.md [hunk 3]
```diff
@@ -29,12 +29,12 @@ A third party metrics collector can be used to monitor the rate of
 change of this metric to determine how the number of files are growing over time.
 
 The number of files in Alluxio impacts the following:
-* Size of heap required by the master - Each file takes approximately 1 - 2 KB. If RocksDB is used,
-most file metadata is stored off-heap, and the size of the heap impacts how many files’ metadata can
-be cached on heap. See the
+* Size of heap required by the master - Each file and its directory structure takes approximately 4KB.
```
Comment (jiacheliu3) [inline docs/en/operation/Scalability-Tuning.md:32]: Do you mean the file plus what this file will incur on the directory equals 4KB? I was first told the size is 1KB 2 years back, then I saw this 1-2KB last year, then I see this 4KB now... I'm a little confused based on the current doc. And probably that implies this  part should be further elaborated. What do you think?

File: docs/en/operation/Scalability-Tuning.md [hunk 4]
```diff
@@ -111,6 +111,8 @@ clients.
 * Amount of network bandwidth required by the worker - We recommend at least 10 MB/s per concurrent
 client. This resource is less important if a majority of tasks have locality and use short circuit.
 
+The metric "Worker.ActiveClients" displays the current number of clients connected to a particular worker,
```
Comment (jiacheliu3) [inline docs/en/operation/Scalability-Tuning.md:114]: I guess it makes sense to mention this is added in version blah. It's not too intuitive for the user to compare the latest doc to the doc for a previous version to note this difference.

File: docs/en/operation/Scalability-Tuning.md [hunk 5]
```diff
@@ -169,10 +180,14 @@ We recommend at least 8 GB of disk space for writing logs. The write speed of th
 least 128 MB/s.
 
 When using embedded journal, the disk space is proportional to the namespace size and typical number
-of write operations within a snapshot period. We recommend at least 8 GB of disk space plus 2 GB for
+of write operations within a snapshot period. We recommend at least 8 GB of disk space plus 8GB for
```
Comment (jiacheliu3) [inline docs/en/operation/Scalability-Tuning.md:183]:
```suggestion
of write operations within a snapshot period. We recommend at least 8 GB of disk space plus 8 GB for
```

Comment (jiacheliu3) [reply docs/en/operation/Scalability-Tuning.md:183]: Or `8GB` should be the preferred compact format?

File: docs/en/operation/Scalability-Tuning.md [hunk 6]
```diff
@@ -139,6 +141,15 @@ ALLUXIO_MASTER_JAVA_OPTS+=" -Xms256g -Xmx256g "
 When setting the heap size, ensure that there is enough memory allocated for off heap storage.
 For example, spawning `4000` threads with a default thread stack size of `1 MB` requires at least
 `4 GB` of off-heap space available.
+* Network buffers are often allocated from a pool of direct memory in Java. 
+The configuration controlling the maximum size of direct memory allocated defaults to the xmx setting,
```
Comment (jiacheliu3) [inline docs/en/operation/Scalability-Tuning.md:145]:
```suggestion
The configuration controlling the maximum size of direct memory allocated defaults to the `-Xmx` setting, 
```

File: docs/en/operation/Scalability-Tuning.md [hunk 7]
```diff
@@ -139,6 +141,15 @@ ALLUXIO_MASTER_JAVA_OPTS+=" -Xms256g -Xmx256g "
 When setting the heap size, ensure that there is enough memory allocated for off heap storage.
 For example, spawning `4000` threads with a default thread stack size of `1 MB` requires at least
 `4 GB` of off-heap space available.
+* Network buffers are often allocated from a pool of direct memory in Java. 
+The configuration controlling the maximum size of direct memory allocated defaults to the xmx setting, 
+which can leave very little space for the other critical processes in the system. 
+We recommend setting it to 10GB for both Alluxio Master and Alluxio Workers in a typical deployment, and only increase it
```
Comment (jiacheliu3) [inline docs/en/operation/Scalability-Tuning.md:147]: I thought big direct memory is only needed for workers as they do large data IOs. Is that the case for the master too?

File: docs/en/operation/Scalability-Tuning.md [hunk 8]
```diff
@@ -139,6 +141,15 @@ ALLUXIO_MASTER_JAVA_OPTS+=" -Xms256g -Xmx256g "
 When setting the heap size, ensure that there is enough memory allocated for off heap storage.
 For example, spawning `4000` threads with a default thread stack size of `1 MB` requires at least
 `4 GB` of off-heap space available.
+* Network buffers are often allocated from a pool of direct memory in Java.
```
Comment (jiacheliu3) [inline docs/en/operation/Scalability-Tuning.md:144]: Do you know when exactly is buffer allocated on/off heap? I mean it sounds like the buffer is not always off heap.

File: docs/en/operation/Performance-Tuning.md [hunk 9]
```diff
@@ -246,6 +259,17 @@ Alluxio workers matches the the number of HDFS datanodes, set
 `alluxio.worker.ufs.instream.cache.max.size=<value of HDFS setting dfs.datanode.handler.count>`
 under the assumption that the workload is spread evenly over Alluxio workers.
 
+## Job Service Tuning
+
+### Job Service Capacity
+Job service limits the total number of currently running jobs to control its resource usage. 
+If jobs tend to be created in large batches, consider increasing `alluxio.job.master.job.capacity` to a larger value than the default 100K.
```
Comment (jiacheliu3) [inline docs/en/operation/Performance-Tuning.md:266]: Do you think it makes sense to mention also the internal capacity for a distributed command https://github.com/Alluxio/alluxio/blob/2f7c61c7cf8cf38cdb0987902e8ffe4bc2d02a2f/shell/src/main/java/alluxio/cli/fs/command/AbstractDistributedJobCommand.java#L33
and also this is possible overriden by an input argument?

Also what happens when the capacity is full? Will new jobs be rejected or blocking?

**Review by jiacheliu3 — COMMENTED** [review]
Left some comments. PTAL, thanks!

File: docs/en/operation/Performance-Tuning.md (hunk 1 above)
Comment (yuzhu) [reply docs/en/operation/Performance-Tuning.md:227]: the prefetch thread pool default is usually large enough to handle all situations, that's why I left it out to reduce confusion.

File: docs/en/operation/Scalability-Tuning.md (hunk 3 above)
Comment (yuzhu) [reply docs/en/operation/Scalability-Tuning.md:32]: the inode itself is still estimated to take 2KB per inode, but we need to consider all the other overhead, like the directory structure, and JVM heap usage. we used 4KB in our latest calculation for Rocksdb cache, that's why i adjusted to 4kb here for consistency.

File: docs/en/operation/Scalability-Tuning.md (hunk 8 above)
Comment (yuzhu) [reply docs/en/operation/Scalability-Tuning.md:144]: I think netty buffers are allocated from direct memory? i am not entirely sure. @ggezer do you have more context here?

File: docs/en/operation/Performance-Tuning.md [hunk 10]
```diff
@@ -246,6 +259,17 @@ Alluxio workers matches the the number of HDFS datanodes, set
...
+## Job Service Tuning
+
+### Job Service Capacity
+Job service limits the total number of currently running jobs to control its resource usage. 
+If jobs tend to be created in large batches, consider increasing `alluxio.job.master.job.capacity` to a larger value than the default 100K. 
+
+### Job Service Throughput
+When there are many concurrent jobs running, and a higher throughput is desired, consider increasing `alluxio.job.worker.threadpool.size` configuration. 
+This allows each job worker to run with more parallel threads. The drawback is it will compete for resources on the worker machines. 
+Recommend 2x core count if most jobs are running in off peak hours only and 1/2 to 1x core count if jobs are running in all hours.
```
Comment (madanadit) [inline docs/en/operation/Performance-Tuning.md:271]: maybe link to the CLI option to set this for a specific job as an alternative

File: docs/en/operation/Performance-Tuning.md (hunk 9 above)
Comment (madanadit) [reply docs/en/operation/Performance-Tuning.md:266]: elaborate what a "job" means here? for example, a single distributedLoad operation can have X jobs controlling the % of the total job server capacity consumed by the single distributedLoad operation.

File: docs/en/operation/Scalability-Tuning.md [hunk 11]
```diff
@@ -29,12 +29,12 @@ A third party metrics collector can be used to monitor the rate of
...
-* Size of heap required by the master - Each file takes approximately 1 - 2 KB. If RocksDB is used,
-most file metadata is stored off-heap, and the size of the heap impacts how many files’ metadata can
-be cached on heap. See the
+* Size of heap required by the master - Each file and its directory structure takes approximately 4KB. 
+If RocksDB is used, most file metadata is stored off-heap, 
+and the size of the heap impacts how many files’ metadata can be cached on the heap. See the
 [RocksDB section]({{ '/en/operation/Metastore.html#rocksdb-metastore' | relativize_url }}) for more
 information.
-* Size of disk required for journal storage - Each file takes approximately 1 - 2 KB on disk.
+* Size of disk required for journal storage - Each file takes approximately 4 KB on disk.
```
Comment (madanadit) [inline docs/en/operation/Scalability-Tuning.md:37]: At peak, we may have 2 snapshots during checkpointing. Do we need to 2x this number for how much local disk needs to be provisioned per master?

**Review by madanadit — COMMENTED** [review]
Thanks a lot @yuzhu! Left a few minor comments.

File: docs/en/operation/Performance-Tuning.md (hunk 2 above)
Comment (yuzhu) [reply docs/en/operation/Performance-Tuning.md:224]: fixed

File: docs/en/operation/Scalability-Tuning.md (hunk 4 above)
Comment (yuzhu) [reply docs/en/operation/Scalability-Tuning.md:114]: it was merged 2 days ago.. so it is hard to put a version next to it.. 2.7?

Comment (jiacheliu3) [reply docs/en/operation/Scalability-Tuning.md:114]: What I used to do (for example https://docs.alluxio.io/os/user/stable/en/deploy/Running-Alluxio-On-Kubernetes.html#basic-setup mentioned `2.3`) is referring to the next version, but now the versioning policy has changed. So might need @apc999 to confirm on the next version and some general guidance on below:

For a user (and oftentimes we are users too, to feature we are not familiar enough about), it's natural to go to the latest feature doc, and later downloading a versioned tarball which may not necessarily match. I think it's helpful to associate a version, at least for features that only exist in the latest version number (`edge` or `SNAPSHOT`). What do you think @apc999 ?

Comment (jiacheliu3) [reply docs/en/operation/Scalability-Tuning.md:114]: What I used to do (for example https://docs.alluxio.io/os/user/stable/en/deploy/Running-Alluxio-On-Kubernetes.html#basic-setup mentioned `2.3`) is referring to the next version, but now the versioning policy has changed. So might need @apc999 to confirm on the next version and some general guidance on below:

For a user (and oftentimes we are users too, to feature we are not familiar enough about), it's natural to go to the latest feature doc, and later downloading a versioned tarball which may not necessarily match. I think it's helpful to associate a version, at least for features that only exist in the latest version number (`edge` or `SNAPSHOT`). What do you think @apc999 ?

**Comment by apc999** [comment]
any update on this PR?

File: docs/en/operation/Scalability-Tuning.md (hunk 4 above)
Comment (yuzhu) [reply docs/en/operation/Scalability-Tuning.md:114]: i think i will remove this comment for now, and insert it once we have a release.

File: docs/en/operation/Scalability-Tuning.md (hunk 7 above)
Comment (yuzhu) [reply docs/en/operation/Scalability-Tuning.md:147]: master can do large data /network IO too? eg pinlist?

File: docs/en/operation/Performance-Tuning.md [hunk 12]
```diff
@@ -214,6 +214,19 @@ Increasing the number of threads can decrease the staleness of the UFS path cach
...
 
+### Metadata Sync
+
+If the content on the UFS are modified without going through Alluxio, Alluxio needs to sync its metadata with the UFS to reflect those changes in Alluxio namespace. 
+The cost of metadata sync scales linearly with the number of files in the directory that is being synced. 
+If metadata sync operation happens frequently on large directories, more threads may be allocated to speed up this process. 
+Two configurations are relevant here. 
+
+'alluxio.master.metadata.sync.concurrency.level' controls the concurrency that is used in a single sync operation.
+Adjust this to 1x to 2x core count on the master node to speed up the speed of metadata sync.
```
[the remark this answers is not in the document]
Comment (yuzhu) [reply docs/en/operation/Performance-Tuning.md:225]: done

File: docs/en/operation/Performance-Tuning.md (hunk 10 above)
Comment (yuzhu) [reply docs/en/operation/Performance-Tuning.md:271]: we don't have that.

**Comment by yuzhu** [comment]
@jiacheliu3 @madanadit PTAL

**Review by madanadit — APPROVED** [review]
Thx!

**Comment by yuzhu** [comment]
alluxio-bot, merge this please

**Pull request merged**